pipelines/push-to-internal.yml

```diff
@@ -0,0 +1,22 @@
+# Do not run this when opening PRs ...
+pr: none
+
+# ... rather, just run this every time a branch is updated:
+trigger:
+  branches:
+    include:
+      - '*'  # Trigger on all branches
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+steps:
+- script: |
+    BRANCH_NAME=`echo $BUILD_SOURCEBRANCH | sed -e 's!^refs/heads/!!'`
+    git remote add internal https://msrcambridge:$(GitHub_FSMol_Sync_PAT)@dev.azure.com/msrcambridge/GenChem/_git/FS-Mol
+    git remote update
+    git checkout -b ${BRANCH_NAME}/merged
+    git reset --hard internal/main
+    git reset --hard internal/${BRANCH_NAME} || true
```

Comment:
I'm curious: why do we need to go to `main` first and then to the target branch, and what is the `|| true` bit doing?

Reply:
Think of the scenario when the branch doesn't exist on the `internal` remote - in that case, we want to create a fresh branch based on the `internal` main and our scenario.
However, if it already exists (from an earlier push, for example), we want to build on top of that, and so we try to reset that state; as that can fail, we add `|| true` to make sure that the pipeline doesn't fail.